block/polka.go

```diff
@@ -46,15 +56,43 @@ func (preVote PreVote) String() string {
 	if preVote.Block != nil {
 		block = preVote.Block.String()
 	}
-	return fmt.Sprintf("PreVote(%s,Round=%d,Height=%d)", block, preVote.Round, preVote.Height)
+	return fmt.Sprintf("PreVote(%s,Round=%d,Height=%d)",
+		block,
+		preVote.Round,
+		preVote.Height)
 }
 
+// SignedPreVote is the signed version of a PreVote
 type SignedPreVote struct {
 	PreVote
 	Signature sig.Signature
 	Signatory sig.Signatory
 }
 
+func (sPreVote SignedPreVote) String() string {
+	return fmt.Sprintf("SignedPreVote(%s,Signature=%v,Signatory=%v)",
+		sPreVote.PreVote.String(),
+		sPreVote.Signature,
+		sPreVote.Signatory)
+}
+
+// GenerateSignedPreVote from a block signed with a random private
+// key
+func GenerateSignedPreVote(block *Block) SignedPreVote {
```

Comment:
This function should be in a test file not in the live codebase. This is a confusing API that directs people to generate signed PreVotes incorrectly. It is intentional that there is no constructor and that you must sign a PreVote to get a SignedPreVote.